sklearn/inspection/_partial_dependence.py

```diff
@@ -5,159 +5,33 @@
 #          Nicolas Hug
 # License: BSD 3 clause
 
-from itertools import chain
 from itertools import count
-import numbers
-from collections.abc import Iterable
-import warnings
 
 import numpy as np
-from scipy import sparse
-from scipy.stats.mstats import mquantiles
-from joblib import Parallel, delayed
 
 from ..base import is_classifier, is_regressor
-from ..pipeline import Pipeline
-from ..utils.extmath import cartesian
-from ..utils import check_array
 from ..utils import check_matplotlib_support  # noqa
 from ..utils import _safe_indexing
-from ..utils import _determine_key_type
 from ..utils import _get_column_indices
-from ..utils.validation import check_is_fitted
-from ..tree._tree import DTYPE
-from ..exceptions import NotFittedError
-from ..ensemble._gb import BaseGradientBoosting
-from sklearn.ensemble._hist_gradient_boosting.gradient_boosting import (
-    BaseHistGradientBoosting)
+from ._base import (
+    _get_predictions, _validate_pdp_ice_parameters, _grid_from_X,
+    _plot_pdp_ice)
 
 
 __all__ = ['partial_dependence', 'plot_partial_dependence',
            'PartialDependenceDisplay']
 
 
-def _grid_from_X(X, percentiles, grid_resolution):
```

Comment:
moved to _base

Reply:
Moved back to _partial_dependence.py as suggested.